Backend-Challenges.md

```diff
@@ -125,10 +272,79 @@ Additionally, most explorers will display the addresses of both the sender and r
+9. Next to your app, click on `View Key`
+10. Copy and paste your Alchemy API key (HTTPS) into the format
+
+After populating the command with your private key and api keys, you can paste it into the terminal and run it! The smart contract will now be deployed via the wallet of which you have provided the private key, and after some time, also verified!
+
+🎉🎉🎉 **Congratulations!!!** 🎉🎉🎉
+You have just built, deployed and verified an NFT smart contract! Take note of the contract address in the terminal output, for example: `Deployed to: 0x.....` , and let's proceed to the next step!
+
+### Smart contract initialization through Etherscan
+
+Once you have your contract address, let's move on to the [Goerli Etherscan](https://goerli.etherscan.io/).
+
+1. Paste in your contract address
+2. Click on `Contract` > `Read Contract`
+3. Click on `tokenURI` and put 0 for `_id`
+
+It will return `0.json`. As you can see, the base URI still needs to be set! Let's fix that.
+
+1. Get your base URI from Pinata for the metadata folder
+2. On Etherscan, click on the next tab `Write Contract`
+3. Click on `Connect to Web3` and follow the steps
+4. Once you're connected, click on `setBaseURI`
+5. Paste in your URI in the `_uri` field (i.e. `ipfs://yourURI/`) and click on `Write`
+6. Your wallet will pop up, asking you to pay a small gas fee for the transaction. Confirm your transaction.
+7. After a while, your transaction will complete. You can view your transaction status by clicking on `View Transaction`
+8. When your transaction is completed, go back to `Read Contract` and check the token URI again. Feel free to paste it into your browser to check out your metadata!
+
+### Minting on Etherscan
+
+Right now, we still have to create the NFTs themselves by minting, so let's go ahead and do that! Of course, since we have only uploaded five metadata files, if you mint more than five, those will not have any metadata.
+
+1. Click on `Write Contract`
+2. Click on `Mint`
+3. Put 0 in the payable amount (Our mints are free)
+4. Put an amount between 0 and 5 (remember, max mint defaults to 5 in the contract)
+5. Click on `Write` and wait for the transaction to complete
+
+### Viewing your NFTs on Opensea
+
+Once you have minted your NFTs, you should be able to view them on Opensea!
+
+1. Let's head over to the [Testnet Opensea](https://testnets.opensea.io/)
+2. In the search bar, paste your contract address
+3. Your collection should appear with the minted NFTs!
+
+If you have managed to make it this far, **give yourself a pat on the back, you've earned it!** Of course, these challenges were just the tip of the iceberg. The NFTs can be extended with loads of functionalities, you can create an Opensea account and edit your collection, you can optimize your smart contracts so the gas fees are less and there are a lot of security measures you could take to make your contract safer, and the list goes on and on.
```

Comment:
I think its good to note that we will be supplying a non-exhaustive list of stretch-challenges they can work on to refine their skills from here on, or start working on an idea of their own